File: backend/scripts/healthcheck.js
// backend/scripts/healthcheck.js
// Used by Docker HEALTHCHECK instruction
const http = require('http');

const options = {
  hostname: 'localhost',
  port:     process.env.PORT || 5000,
  path:     '/health',
  method:   'GET',
  timeout:  3000,
};

const req = http.request(options, (res) => {
  if (res.statusCode === 200) {
    process.exit(0); // healthy
  } else {
    process.exit(1); // unhealthy
  }
});

req.on('error',   () => process.exit(1));
req.on('timeout', () => process.exit(1));
req.end();

File: ai-service/scripts/healthcheck.py
# ai-service/scripts/healthcheck.py
import sys
import urllib.request

try:
    url = f"http://localhost:{__import__('os').environ.get('PORT', 8000)}/health"
    req = urllib.request.urlopen(url, timeout=3)
    if req.status == 200:
        sys.exit(0)
    else:
        sys.exit(1)
except Exception:
    sys.exit(1)